### PR TITLE
Connect the equality solver to theory arith

### PR DESCRIPTION
**Pull request by ajreynol**
`--arith-eq-solver` is a new option to enable the equality solver in arithmetic, disabled by default.

This PR connects the equality solver to TheoryArith when this option is enabled. 

This is in preparation for the central equality engine.

### CODE REVIEW

File: src/theory/arith/theory_arith.cpp [hunk 1]
```diff
@@ -73,6 +80,10 @@ ProofRuleChecker* TheoryArith::getProofChecker()
 
 bool TheoryArith::needsEqualityEngine(EeSetupInfo& esi)
 {
+  if (d_eqSolver != nullptr)
+  {
+    return d_eqSolver->needsEqualityEngine(esi);
```
Comment (nafur) [inline src/theory/arith/theory_arith.cpp:85]: It looks a bit weird, that `d_internal` isn't even considered when `d_eqSolver` is used. It is correct, though, as `d_eqSolver` always returns `true`.
I'd still suggest to either formulate it like
`if (d_eqSolver != nullptr && d_eqSolver->needsEqualityEngine(esi)) return true` or just say `if (d_eqSolver != nullptr) return true;` to make this piece of code easier to understand.

**Review by nafur — COMMENTED** [review]
Just one comment to make the code a bit easier to understand.

File: src/theory/arith/theory_arith.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/arith/theory_arith.cpp:85]: The purpose of this method is to initialize esi; I'll comment on why the control flow looks like this.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**